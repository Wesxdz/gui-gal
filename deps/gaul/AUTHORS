Primary author/maintainer: "Stewart Adcock" <stewart@linux-domain.com>

Stewart is grateful to all users who have provided comments, compilation reports, bug reports, patches and donations (not necessarily in that order).  While this is not an exhaustive list, he would like to specifically thank the following people:

Prior to release 0.1850:
Andrew W. Nosenko - Provided a large set of excellent patches to fix a multitude of mostly minor, but not insignificant, problems.
Al Cripps - Constructive bug reports with suggested fixes.

Prior to release 0.1849:
Al Cripps - Suggested fixes and compilation instructions for MS VC++.
Marie-Pierre Brungard - Patch for building DLLs.
Johannes Zellner - Constructive bug reports.
Andrew W. Nosenko - sincos() typo patch.

Prior to release 0.1848:
Al Cripps - Suggested fix in memory_chunks.c for MS VC++.

Prior to release 0.1847:
Dave Wilkinson - Compilation reports for IRIX and Solaris.
Syrrx, Inc. - More dollars :-)
CTBP (http://ctbp.ucsd.edu/) - Computing resources.
SGI, Inc. - Access to an Altix 3200 for porting and benchmarking.

Prior to release 0.1846:
Ying-Chieh Liao - Patch for stdbool.h include on FreeBSD.

Prior to release 0.1845:
Gad Abraham - Testing Debian package.
Henk van Lingen - ga_crossover_bitstring_doublepoints() and ga_bit_copy() bug reports.

Prior to release 0.1843:
Kalin Kozhuharov - HP-UX/MPP and Hitachi SR8000 testing and compilation reports.

Prior to release 0.1842:
McKenzie Keith - Providing Cygwin binaries.  Reporting MingW issues.
Ben Webb - Tesing Cygwin binaries.
Kate Mullen - Simulated annealling documentation.

Prior to release 0.1841:
Daniel Pick - Bug reports and fix.
McKenzie Keith - Windows DLL compilation.

Prior to release 0.1840:
Antti Akonniemi - MacOS X compilation report.
Mark T. Richardson - QNX V6 compilation report.

Prior to release 0.1839:
Syrrx, Inc. - Dollars :-)

Prior to release 0.1838:
Stepan Harbek - Cygwin compilation reports and tips.
Sourceforge (http://sourceforge.net/) - Hosting services.

