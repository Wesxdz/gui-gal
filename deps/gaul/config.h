/* config.h.  Generated by configure.  */
/* config.h.in.  Generated from configure.in by autoheader.  */

/* Overall debugging level. */
#define DEBUG 0

/* Record date of compilation. */
#define GA_BUILD_DATE_STRING "23/12/21"

/* Core GAUL debugging level. */
#define GA_DEBUG 0

/* Default number of processes. */
#define GA_DEFAULT_NUM_PROCESSES 8

/* Default number of threads. */
#define GA_DEFAULT_NUM_THREADS 4

/* Minor version number for libtool. */
#define GA_LT_MINOR_VERSION 50

/* Major version number. */
#define GA_MAJOR_VERSION 0

/* Minor version number. */
#define GA_MINOR_VERSION 1850

/* Environment variable used to request processes. */
#define GA_NUM_PROCESSES_ENVVAR_STRING "GA_NUM_PROCESSES"

/* Environment variable used to request threads. */
#define GA_NUM_THREADS_ENVVAR_STRING "GA_NUM_THREADS"

/* Patch level. */
#define GA_PATCH_VERSION 0

/* Record compilation platform information. */
#define GA_UNAME_STRING "Linux heonae 5.4.0-91-generic #102-Ubuntu SMP Fri Nov 5 16:31:28 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux"

/* Version string. */
#define GA_VERSION_STRING "0.1850-0"

/* Whether to dump certian statistics. */
#define GA_WRITE_STATS 0

/* Whether bcmp() is available */
#define HAVE_BCMP 1

/* Whether bcopy() is available */
#define HAVE_BCOPY 1

/* Define to 1 if you have the <dlfcn.h> header file. */
#define HAVE_DLFCN_H 1

/* Define to 1 if you have the `gethostname' function. */
#define HAVE_GETHOSTNAME 1

/* Define to 1 if you have the `index' function. */
#define HAVE_INDEX 1

/* Define to 1 if you have the <inttypes.h> header file. */
#define HAVE_INTTYPES_H 1

/* Define to 1 if you have the `m' library (-lm). */
/* #undef HAVE_LIBM */

/* Define to 1 if you have the `mpi' library (-lmpi). */
/* #undef HAVE_LIBMPI */

/* Define to 1 if you have the `mpich' library (-lmpich). */
/* #undef HAVE_LIBMPICH */

/* Define to 1 if you have the <limits.h> header file. */
#define HAVE_LIMITS_H 1

/* Define to 1 if you have the `max' function. */
/* #undef HAVE_MAX */

/* Define to 1 if you have the `memcmp' function. */
#define HAVE_MEMCMP 1

/* Define to 1 if you have the `memcpy' function. */
#define HAVE_MEMCPY 1

/* Define to 1 if you have the <memory.h> header file. */
#define HAVE_MEMORY_H 1

/* Whether memset() is available */
#define HAVE_MEMSET 1

/* Define to 1 if you have the `min' function. */
/* #undef HAVE_MIN */

/* Whether to use MPI for parallel code. */
/* #undef HAVE_MPI */

/* Whether libpthread is installed. */
#define HAVE_PTHREADS 1

/* Define to 1 if you have the <pwd.h> header file. */
#define HAVE_PWD_H 1

/* Define to 1 if you have the `sincos' function. */
#define HAVE_SINCOS 1

/* Whether S-Lang is installed. */
/* #undef HAVE_SLANG */

/* Define to 1 if you have the `snooze' function. */
/* #undef HAVE_SNOOZE */

/* Define if you have a working <stdbool.h> header file. */
#define HAVE_STDBOOL_H 1

/* Define to 1 if you have the <stdint.h> header file. */
#define HAVE_STDINT_H 1

/* Define to 1 if you have the <stdlib.h> header file. */
#define HAVE_STDLIB_H 1

/* Define to 1 if you have the `strcasecmp' function. */
#define HAVE_STRCASECMP 1

/* Define to 1 if you have the `strchr' function. */
#define HAVE_STRCHR 1

/* Define to 1 if you have the `strcmp' function. */
#define HAVE_STRCMP 1

/* Define to 1 if you have the `strcpy' function. */
#define HAVE_STRCPY 1

/* Define to 1 if you have the `strdup' function. */
#define HAVE_STRDUP 1

/* Define to 1 if you have the <strings.h> header file. */
#define HAVE_STRINGS_H 1

/* Define to 1 if you have the <string.h> header file. */
#define HAVE_STRING_H 1

/* Whether strlen() is available */
#define HAVE_STRLEN 1

/* Define to 1 if you have the `strncasecmp' function. */
#define HAVE_STRNCASECMP 1

/* Whether strncmp() is available */
#define HAVE_STRNCMP 1

/* Whether strncpy() is available */
#define HAVE_STRNCPY 1

/* Define to 1 if you have the `strndup' function. */
#define HAVE_STRNDUP 1

/* Define to 1 if you have the `strpbrk' function. */
#define HAVE_STRPBRK 1

/* Define to 1 if you have the `strsep' function. */
#define HAVE_STRSEP 1

/* Define to 1 if you have the <sys/select.h> header file. */
#define HAVE_SYS_SELECT_H 1

/* Define to 1 if you have the <sys/signal.h> header file. */
#define HAVE_SYS_SIGNAL_H 1

/* Define to 1 if you have the <sys/stat.h> header file. */
#define HAVE_SYS_STAT_H 1

/* Define to 1 if you have the <sys/time.h> header file. */
#define HAVE_SYS_TIME_H 1

/* Define to 1 if you have the <sys/types.h> header file. */
#define HAVE_SYS_TYPES_H 1

/* Define to 1 if you have the <sys/wait.h> header file. */
#define HAVE_SYS_WAIT_H 1

/* Define to 1 if you have the <unistd.h> header file. */
#define HAVE_UNISTD_H 1

/* Whether usleep() is available */
#define HAVE_USLEEP 1

/* Define to 1 if you have the `waitpid' function. */
#define HAVE_WAITPID 1

/* Define if compiler has built-in _Bool type. */
#define HAVE__BOOL 1

/* Memory alignment boundry characteristics. */
#define MEMORY_ALIGN_SIZE 8

/* Whether to use debugging memory allocation routines. */
/* #undef MEMORY_ALLOC_DEBUG */

/* Whether to use safe memory allocation routines. */
#define MEMORY_ALLOC_SAFE 1

/* Define to disable memory chunk allocation routines. */
/* #undef MEMORY_CHUNKS_MIMIC */

/* Whether to pad memory. */
/* #undef MEMORY_PADDING */

/* Define to the address where bug reports for this package should be sent. */
#define PACKAGE_BUGREPORT "gaul@linux-domain.com"

/* Define to the full name of this package. */
#define PACKAGE_NAME "gaul-devel"

/* Define to the full name and version of this package. */
#define PACKAGE_STRING "gaul-devel 0.1850-0"

/* Define to the one symbol short name of this package. */
#define PACKAGE_TARNAME "gaul-devel"

/* Define to the version of this package. */
#define PACKAGE_VERSION "0.1850-0"

/* The size of a `char', as computed by sizeof. */
#define SIZEOF_CHAR 1

/* The size of a `double', as computed by sizeof. */
#define SIZEOF_DOUBLE 8

/* The size of a `int', as computed by sizeof. */
#define SIZEOF_INT 4

/* The size of a `long', as computed by sizeof. */
#define SIZEOF_LONG 8

/* The size of a `void *', as computed by sizeof. */
#define SIZEOF_VOID_P 8

/* Define to 1 if you have the ANSI C header files. */
#define STDC_HEADERS 1

/* Define if using Sun's Forte C Compiler */
#define SUN_FORTE_C 0

/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
#define TIME_WITH_SYS_TIME 1

/* Define to 1 if your <sys/time.h> declares `struct tm'. */
/* #undef TM_IN_SYS_TIME */

/* Whether to use memory chunk handling for chromosomes. */
/* #undef USE_CHROMO_CHUNKS */

/* Whether OpenMP should be used. */
/* #undef USE_OPENMP */

/* Whether to cripple GAUL so it works on windows. */
/* #undef W32_CRIPPLED */

/* Define to empty if `const' does not conform to ANSI C. */
/* #undef const */
