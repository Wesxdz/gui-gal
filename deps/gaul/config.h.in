/* config.h.in.  Generated from configure.in by autoheader.  */

/* Overall debugging level. */
#undef DEBUG

/* Record date of compilation. */
#undef GA_BUILD_DATE_STRING

/* Core GAUL debugging level. */
#undef GA_DEBUG

/* Default number of processes. */
#undef GA_DEFAULT_NUM_PROCESSES

/* Default number of threads. */
#undef GA_DEFAULT_NUM_THREADS

/* Minor version number for libtool. */
#undef GA_LT_MINOR_VERSION

/* Major version number. */
#undef GA_MAJOR_VERSION

/* Minor version number. */
#undef GA_MINOR_VERSION

/* Environment variable used to request processes. */
#undef GA_NUM_PROCESSES_ENVVAR_STRING

/* Environment variable used to request threads. */
#undef GA_NUM_THREADS_ENVVAR_STRING

/* Patch level. */
#undef GA_PATCH_VERSION

/* Record compilation platform information. */
#undef GA_UNAME_STRING

/* Version string. */
#undef GA_VERSION_STRING

/* Whether to dump certian statistics. */
#undef GA_WRITE_STATS

/* Whether bcmp() is available */
#undef HAVE_BCMP

/* Whether bcopy() is available */
#undef HAVE_BCOPY

/* Define to 1 if you have the <dlfcn.h> header file. */
#undef HAVE_DLFCN_H

/* Define to 1 if you have the `gethostname' function. */
#undef HAVE_GETHOSTNAME

/* Define to 1 if you have the `index' function. */
#undef HAVE_INDEX

/* Define to 1 if you have the <inttypes.h> header file. */
#undef HAVE_INTTYPES_H

/* Define to 1 if you have the `m' library (-lm). */
#undef HAVE_LIBM

/* Define to 1 if you have the `mpi' library (-lmpi). */
#undef HAVE_LIBMPI

/* Define to 1 if you have the `mpich' library (-lmpich). */
#undef HAVE_LIBMPICH

/* Define to 1 if you have the <limits.h> header file. */
#undef HAVE_LIMITS_H

/* Define to 1 if you have the `max' function. */
#undef HAVE_MAX

/* Define to 1 if you have the `memcmp' function. */
#undef HAVE_MEMCMP

/* Define to 1 if you have the `memcpy' function. */
#undef HAVE_MEMCPY

/* Define to 1 if you have the <memory.h> header file. */
#undef HAVE_MEMORY_H

/* Whether memset() is available */
#undef HAVE_MEMSET

/* Define to 1 if you have the `min' function. */
#undef HAVE_MIN

/* Whether to use MPI for parallel code. */
#undef HAVE_MPI

/* Whether libpthread is installed. */
#undef HAVE_PTHREADS

/* Define to 1 if you have the <pwd.h> header file. */
#undef HAVE_PWD_H

/* Define to 1 if you have the `sincos' function. */
#undef HAVE_SINCOS

/* Whether S-Lang is installed. */
#undef HAVE_SLANG

/* Define to 1 if you have the `snooze' function. */
#undef HAVE_SNOOZE

/* Define if you have a working <stdbool.h> header file. */
#undef HAVE_STDBOOL_H

/* Define to 1 if you have the <stdint.h> header file. */
#undef HAVE_STDINT_H

/* Define to 1 if you have the <stdlib.h> header file. */
#undef HAVE_STDLIB_H

/* Define to 1 if you have the `strcasecmp' function. */
#undef HAVE_STRCASECMP

/* Define to 1 if you have the `strchr' function. */
#undef HAVE_STRCHR

/* Define to 1 if you have the `strcmp' function. */
#undef HAVE_STRCMP

/* Define to 1 if you have the `strcpy' function. */
#undef HAVE_STRCPY

/* Define to 1 if you have the `strdup' function. */
#undef HAVE_STRDUP

/* Define to 1 if you have the <strings.h> header file. */
#undef HAVE_STRINGS_H

/* Define to 1 if you have the <string.h> header file. */
#undef HAVE_STRING_H

/* Whether strlen() is available */
#undef HAVE_STRLEN

/* Define to 1 if you have the `strncasecmp' function. */
#undef HAVE_STRNCASECMP

/* Whether strncmp() is available */
#undef HAVE_STRNCMP

/* Whether strncpy() is available */
#undef HAVE_STRNCPY

/* Define to 1 if you have the `strndup' function. */
#undef HAVE_STRNDUP

/* Define to 1 if you have the `strpbrk' function. */
#undef HAVE_STRPBRK

/* Define to 1 if you have the `strsep' function. */
#undef HAVE_STRSEP

/* Define to 1 if you have the <sys/select.h> header file. */
#undef HAVE_SYS_SELECT_H

/* Define to 1 if you have the <sys/signal.h> header file. */
#undef HAVE_SYS_SIGNAL_H

/* Define to 1 if you have the <sys/stat.h> header file. */
#undef HAVE_SYS_STAT_H

/* Define to 1 if you have the <sys/time.h> header file. */
#undef HAVE_SYS_TIME_H

/* Define to 1 if you have the <sys/types.h> header file. */
#undef HAVE_SYS_TYPES_H

/* Define to 1 if you have the <sys/wait.h> header file. */
#undef HAVE_SYS_WAIT_H

/* Define to 1 if you have the <unistd.h> header file. */
#undef HAVE_UNISTD_H

/* Whether usleep() is available */
#undef HAVE_USLEEP

/* Define to 1 if you have the `waitpid' function. */
#undef HAVE_WAITPID

/* Define if compiler has built-in _Bool type. */
#undef HAVE__BOOL

/* Memory alignment boundry characteristics. */
#undef MEMORY_ALIGN_SIZE

/* Whether to use debugging memory allocation routines. */
#undef MEMORY_ALLOC_DEBUG

/* Whether to use safe memory allocation routines. */
#undef MEMORY_ALLOC_SAFE

/* Define to disable memory chunk allocation routines. */
#undef MEMORY_CHUNKS_MIMIC

/* Whether to pad memory. */
#undef MEMORY_PADDING

/* Define to the address where bug reports for this package should be sent. */
#undef PACKAGE_BUGREPORT

/* Define to the full name of this package. */
#undef PACKAGE_NAME

/* Define to the full name and version of this package. */
#undef PACKAGE_STRING

/* Define to the one symbol short name of this package. */
#undef PACKAGE_TARNAME

/* Define to the version of this package. */
#undef PACKAGE_VERSION

/* The size of a `char', as computed by sizeof. */
#undef SIZEOF_CHAR

/* The size of a `double', as computed by sizeof. */
#undef SIZEOF_DOUBLE

/* The size of a `int', as computed by sizeof. */
#undef SIZEOF_INT

/* The size of a `long', as computed by sizeof. */
#undef SIZEOF_LONG

/* The size of a `void *', as computed by sizeof. */
#undef SIZEOF_VOID_P

/* Define to 1 if you have the ANSI C header files. */
#undef STDC_HEADERS

/* Define if using Sun's Forte C Compiler */
#undef SUN_FORTE_C

/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
#undef TIME_WITH_SYS_TIME

/* Define to 1 if your <sys/time.h> declares `struct tm'. */
#undef TM_IN_SYS_TIME

/* Whether to use memory chunk handling for chromosomes. */
#undef USE_CHROMO_CHUNKS

/* Whether OpenMP should be used. */
#undef USE_OPENMP

/* Whether to cripple GAUL so it works on windows. */
#undef W32_CRIPPLED

/* Define to empty if `const' does not conform to ANSI C. */
#undef const
