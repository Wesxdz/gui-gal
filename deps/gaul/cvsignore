config.cache
config.log
config.status
stamp-h
stamp-h1
undep.sh
.libs
*/.libs
.deps
*/.deps
*.orig
*/*.orig
*/*/*.orig
*.rej
*/*.rej
*/*/*.rej
examples/diagnostics
examples/pingpong
examples/pingpong_sa
examples/pingpong_tabu
examples/pingpong_tabu2
examples/wildfire
examples/wildfire_forked
examples/goldberg1
examples/goldberg2
examples/royalroad
examples/struggle
examples/struggle_ss
examples/struggle_forked
examples/struggle2
examples/struggle3
examples/struggle4
examples/struggle5
examples/struggle5_mp
examples/nnevolve
gmon.out
*.dat
*.log
*.gz
*.tgz
cvs2cl.pl
*.o
*.lo
*.a
*.la
*.il

