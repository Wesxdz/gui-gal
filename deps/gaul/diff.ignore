*~
.#*
*.a
*.la
*.o
*.lo
*.bak
*.deps
*.dat
*.gmo
*.log
*.orig
*.pot
*.rej
*.sav
*.tmp
*.tgz
*.gz
.deps
.libs
core
core.*
aclocal.m4
configure
config.log
config.h
config.h.in
config.cache
config.status
undep.sh
CVS
TAGS
Makefile
Makefile.in
stamp-h
stamp-h.in
stamp-cat-id
wildfire
pingpong
goldberg1
goldberg2
royalroad
struggle
struggle2
struggle3
struggle4
struggle5
struggle5_mp
struggle_ss
nnevolve
gmon.out
config.log
config.cache
ltmain.sh
