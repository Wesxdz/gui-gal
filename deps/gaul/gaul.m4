dnl ***************************************************************
dnl gaul/gaul.m4
dnl ***************************************************************
dnl
dnl GAUL - Genetic Algorithm Utility Library.
dnl Copyright Â©2001-2003, Stewart Adcock <stewart@linux-domain.com>
dnl All rights reserved.
dnl
dnl The latest version of this program should be available at:
dnl http://www.stewart-adcock.co.uk/
dnl
dnl This program is free software; you can redistribute it and/or modify
dnl it under the terms of the GNU General Public License as published by
dnl the Free Software Foundation; either version 2 of the License, or
dnl (at your option) any later version.  Alternatively, if your project
dnl is incompatible with the GPL, I will probably agree to requests
dnl for permission to use the terms of any other license.
dnl
dnl This program is distributed in the hope that it will be useful, but
dnl WITHOUT ANY WARRANTY WHATSOEVER.
dnl
dnl A full copy of the GNU General Public License should be in the file
dnl "COPYING" provided with this distribution; if not, see:
dnl http://www.gnu.org/
dnl
dnl ***************************************************************
dnl
dnl Some useful m4 macros for GAUL configuration.
dnl
dnl ***************************************************************

dnl GAUL_MEMORY_ALIGN_SIZE
dnl Determine memory alignment characteristics.
AC_DEFUN([GAUL_MEMORY_ALIGN_SIZE],
[ AC_MSG_CHECKING(memory alignment)
AC_CACHE_VAL(MEMORY_ALIGN_SIZE,
[AC_TRY_RUN([
#include <stdio.h>
#include <stdlib.h>
int main(void)
  {
  FILE *f=fopen("conftestval", "w");
  if (!f) exit(1);
  fprintf(f, "%d\n", (unsigned int) sizeof(long)>sizeof(void *)?sizeof(long):sizeof(void *));
  exit(0);
  }
], MEM_ALIGN_SIZE=`cat conftestval`, MEM_ALIGN_SIZE=4, MEM_ALIGN_SIZE=8)])
AC_MSG_RESULT(${MEM_ALIGN_SIZE})
AC_DEFINE_UNQUOTED(MEMORY_ALIGN_SIZE, ${MEM_ALIGN_SIZE}, [Memory alignment boundry characteristics.])
])

dnl GAUL_GNUMAKE_WARN
dnl Produce warning message if non-GNU make is detected.
AC_DEFUN([GAUL_GNUMAKE_WARN],
[
temp=`make -version -f \dev\null 2>&1 | grep GNU`
if test "x$temp" = "x"; then
  AC_MSG_WARN([You are advised to use GNU Make when compiling GAUL])
fi
])

dnl GAUL_DETECT_BOOL
dnl Determine native boolean stuff.
dnl 1) Check whether stdbool.h defines bool and true/false correctly.
dnl 2) Check whether compiler has built-in _Bool type.
AC_DEFUN([GAUL_DETECT_BOOL],
[
AC_CACHE_CHECK([for working stdbool.h],
	   gaul_header_stdbool_h,
	   [AC_TRY_COMPILE([#include <stdbool.h>],
	[bool blah = false;],
	gaul_header_stdbool_h="yes", gaul_header_stdbool_h="no")])
if test "x$gaul_header_stdbool_h" = "xyes"; then
  AC_DEFINE(HAVE_STDBOOL_H, 1, [Define if you have a working <stdbool.h> header file.])
fi
AC_CACHE_CHECK([for built-in _Bool type],
	      gaul__bool,
	     [AC_TRY_COMPILE(,
	     [_Bool blah = (_Bool)0;],
	     gaul__bool="yes", gaul__bool="no")])
if test "x$gaul__bool" = "xno" && test "x$gaul_header_stdbool_h" = "xyes"; then
  AC_TRY_COMPILE([#include <stdbool.h>],
           [_Bool blah = (_Bool)0;],
           gaul__bool="yes", gaul__bool="no")
fi
if test "x$gaul__bool" = "xyes"; then
  AC_DEFINE(HAVE__BOOL, 1, [Define if compiler has built-in _Bool type.])
fi
])


