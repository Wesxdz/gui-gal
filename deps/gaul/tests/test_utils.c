/**********************************************************************
  test_utils.c
 **********************************************************************

  test_utils - Test GAUL general support functions.
  Copyright Â©2003, Stewart Adcock <stewart@linux-domain.com>
  All rights reserved.

  The latest version of this program should be available at:
  http://gaul.sourceforge.net/

  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 2 of the License, or
  (at your option) any later version.  Alternatively, if your project
  is incompatible with the GPL, I will probably agree to requests
  for permission to use the terms of any other license.

  This program is distributed in the hope that it will be useful, but
  WITHOUT ANY WARRANTY WHATSOEVER.

  A full copy of the GNU General Public License should be in the file
  "COPYING" provided with this distribution; if not, see:
  http://www.gnu.org/

 **********************************************************************

  Synopsis:	Test GAUL general support functions.

 **********************************************************************/

/*
 * Includes
 */
#include "gaul.h"

/**********************************************************************
  main()
  synopsis:	Test GAUL's general support code.
  parameters:
  return:
  updated:	25 Jul 2003
 **********************************************************************/

int main(int argc, char **argv)
  {

/*
 * Run utility test functions.
 */
  avltree_test();
  table_test();
  linkedlist_test();

  exit(EXIT_SUCCESS);
  }


