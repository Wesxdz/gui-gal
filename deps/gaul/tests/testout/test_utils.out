Testing my dodgy AVL tree routines.
height: 7
num nodes: 62
tree: 0 1 2 3 4 5 6 7 8 9 A B C D E F G H I J K L M N O P Q R S T U V W X Y Z a b c d e f g h i j k l m n o p q r s t u v w x y z 
tree to 'S' then foo: 0 1 2 3 4 5 6 7 8 9 A B C D E F G H I J K L M N O P Q R S foo 
height: 6
num nodes: 36
tree: 0 1 2 3 4 5 6 7 8 9 a b c d e f g h i j k l m n o p q r s t u v w x y z 
Lookup for 'x': Found 'x'
Lookup for 'X': Not found.
Tests:         PASSED
Testing my table routines.
FIXME: Actually add some tests!
Checking doubly linked lists...
0 1 2 3 4 5 6 7 8 9 
9 8 7 6 5 4 3 2 1 0 
ok
Checking singly linked lists...
0 1 2 3 4 5 6 7 8 9 
9 8 7 6 5 4 3 2 1 0 
ok
